uint32 pm_id
PredicateInfo[] map
